tests/deploy/e2e/test_autostarted.py

```diff
@@ -233,31 +233,23 @@ async def test_deploy_no_code(resource_container, client, clienthelper, environm
         }
     ]
 
-    await clienthelper.put_version_simple(resources, version)
-
-    await clienthelper.wait_for_deployed(version)
-    # The resource state and its logs are not set atomically. This call prevents a race condition.
-    await wait_until_logs_are_available(client, environment, resource_id, expect_nr_of_logs=3)
-
-    response = await client.get_resource(environment, resource_id, logs=True)
-    assert response.code == 200
-    result = response.result
-    assert result["resource"]["status"] == "unavailable"
+    await clienthelper.put_version_simple(resources, version, wait_for_released=True)
 
-    logging.getLogger(__name__).warning("Found results: %s", json.dumps(result["logs"], indent=1))
+    async def log_any() -> bool:
+        response = await client.get_resource(environment, resource_id, logs=True)
+        assert response.code == 200
+        result = response.result
+        logging.getLogger(__name__).warning("Found results: %s", json.dumps(result["logs"], indent=1))
```

Comment:
```suggestion
```

Reply:
it is very handy when the testcase fails,....